--- ../../../../org/Amoeba.java	2020-04-09 11:38:31.030500000 +0200
+++ Amoeba.java	2020-04-21 16:11:45.232000000 +0200
@@ -17,10 +17,10 @@
  */
 public class Amoeba {
     final double ftol;
-    int nfunc;
+    public int nfunc;
     int mpts;
     int ndim;
-    double fmin;
+    public double fmin;
     double[] y;
     double[][] p;
 
@@ -98,7 +98,11 @@
                 return pmin;
             }
             if (nfunc >= NMAX) {
-                throw new Error("NMAX exceeded");
+//                throw new Error("NMAX exceeded");
+                System.out.printf("NMAX (%d) exceeded, contract around best\n", nfunc);
+                y[ihi] = y[ilo]; //contract around ilo
+                if (ndim >= 0) System.arraycopy(p[ilo], 0, p[ihi], 0, ndim);
+                get_psum(p, psum);
             }
             nfunc += 2;
             double ytry = amotry(p, y, psum, ihi, -1.0, func);
@@ -148,4 +152,5 @@
         }
         return ytry;
     }
+
 }
